shopinvader_api_payment_provider_sips/models/payment_transaction_sips.py

```diff
@@ -0,0 +1,76 @@
+# Copyright Odoo SA (https://odoo.com)
+# Copyright 2024 ACSONE SA (https://acsone.eu).
+# @author Stéphane Bidoul <stephane.bidoul@acsone.eu>
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import json
+from urllib.parse import urljoin
+
+from odoo import models
+
+from odoo.addons.payment import utils as payment_utils
+from odoo.addons.payment_sips.models.const import SUPPORTED_CURRENCIES
+
+
+class PaymentTransactionSips(models.Model):
+    _inherit = "payment.transaction"
+
+    def _get_specific_rendering_values(self, processing_values):
+        """Override of payment to return Sips-specific rendering values.
+
+        Note: self.ensure_one() from `_get_processing_values`
+
+        :param dict processing_values: The generic and specific
+        processing values of the transaction
+        :return: The dict of provider-specific processing values
+        :rtype: dict
+        """
+        shopinvader_api_payment = self.env.context.get("shopinvader_api_payment")
+
+        if self.provider_code != "sips" or not shopinvader_api_payment:
+            return super()._get_specific_rendering_values(processing_values)
+
+        shopinvader_api_payment_frontend_redirect_url = self.env.context.get(
+            "shopinvader_api_payment_frontend_redirect_url"
+        )
+        shopinvader_api_payment_base_url = self.env.context.get(
+            "shopinvader_api_payment_base_url"
+        )
+
+        # !!! code copied from Odoo payment_sips module, to use different return and
```

Comment:
Good idea, done in a new commit.

I've had a look at Stripe too, but it doesn't seem it would be possible there, as calling `super()` would already create a request to Stripe, that we couldn't modify. What do you think?

Reply:
Something along these lines:
```python
        shopinvader_api_payment = self.env.context.get("shopinvader_api_payment")

        if self.provider_code != "sips" or not shopinvader_api_payment:
            return super()._get_specific_rendering_values(processing_values)

        shopinvader_api_payment_frontend_redirect_url = self.env.context.get(
            "shopinvader_api_payment_frontend_redirect_url"
        )
        shopinvader_api_payment_base_url = self.env.context.get(
            "shopinvader_api_payment_base_url"
        )

        rv = super().sips_form_generate_values(processing_values)

        base_url = self.get_base_url()
        old_normal_return_url = urljoin(base_url, SipsController._return_url)
        old_automatic_response_url = urljoin(base_url, SipsController._webhook_url)
        normal_return_url = urljoin(shopinvader_api_payment_base_url, "sips/return")
        automatic_response_url = urljoin(
            shopinvader_api_payment_base_url, "sips/webhook"
        )
        rv["Data"] = (
            rv["Data"]
            .replace(old_normal_return_url, normal_return_url)
            .replace(old_automatic_response_url, automatic_response_url)
        )
        rv["Data"] = re.sub(
            r"returnContext=[^|]+",
            "returnContext=%s"
            % json.dumps(
                dict(
                    reference=self.reference,
                    frontend_redirect_url=shopinvader_api_payment_frontend_redirect_url,
                )
            ),
            rv["Data"],
        )
        rv["Seal"] = self._sips_generate_shasign(rv["Data"])

        return rv
```

(Untested, I am currently backporting to v14.0 so there may be mistakes here)

Reply:
Oops I just saw your commit, nevermind :)

Reply:
Yes, I think this is done for Sips :)
Was wondering for Stripe